doc/tools/make_rst.py

```diff
@@ -1423,12 +1659,96 @@ def rstize_text(
 
     if tag_depth > 0:
         print_error(
-            "{}.xml: Tag depth mismatch: too many (or too little) open/close tags.".format(state.current_class), state
+            "{}.xml: Tag depth mismatch: too many (or too little) open/close tags in {}.".format(
+                state.current_class, context_name
+            ),
+            state,
         )
 
     return text
 
 
+def format_context_name(
+    context: Union[ClassDef, SignalDef, ConstantDef, AnnotationDef, PropertyDef, MethodDef, ThemeItemDef, None]
```

Comment:
I wonder if, at this point, it makes sense to create a `Definition` class with just a `definition_name` and a `name` and inherit it in the others? Then shorten/get rid of these Unions.

It would make the intent clearer IMHO.

Reply:
Yeah, I was thinking about this when I added the `param` tag. I think I didn't bother with this because my Python knowledge is limited, but I can look into it.